### PR TITLE
ECAL esConsumes migration of CaloOnlineTools/EcalTools

### PR DESCRIPTION
**Pull request by thomreis**
#### PR description:

This PR migrates ECAL related code in the CaloOnlineTools/EcalTools packages to esConsumes (#31061 ). In addition `getByLabel()` calls were replaced by `getByToken()` and legacy module types were migrated to multithreaded module types.

#### PR validation:

Code compiles.

### CODE REVIEW

File: CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc [hunk 1]
```diff
@@ -1007,7 +1005,7 @@ void EcalCosmicsHists::analyze(edm::Event const& iEvent, edm::EventSetup const&
 
   // get reco tracks
   edm::Handle<reco::TrackCollection> recoTracks;
-  iEvent.getByLabel("cosmicMuons", recoTracks);
+  iEvent.getByToken(tracksToken_, recoTracks);
 
   if (recoTracks.isValid()) {
     //    LogWarning("EcalCosmicsHists") << "... Valid TrackAssociator recoTracks !!! " << recoTracks.product()->size();
```
Comment (tvami) [inline CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc:1011]: Please remove commented out code

File: CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc [hunk 2]
```diff
@@ -1176,7 +1174,7 @@ void EcalCosmicsHists::analyze(edm::Event const& iEvent, edm::EventSetup const&
 
   // Study on Tracks for High Energy events
   edm::Handle<reco::TrackCollection> recoTracksBarrel;
-  iEvent.getByLabel("cosmicMuonsBarrelOnly", recoTracksBarrel);
+  iEvent.getByToken(tracksBarrelToken_, recoTracksBarrel);
   if (!recoTracksBarrel.isValid()) {
     //edm::LogWarning("EcalCosmicsHists") << "RecoTracksBarrel not valid!! " ;
```
Comment (tvami) [inline CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc:1179]: Please remove commented out code

File: CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc [hunk 3]
```diff
@@ -1259,13 +1257,13 @@ void EcalCosmicsHists::analyze(edm::Event const& iEvent, edm::EventSetup const&
...
   edm::Handle<HFRecHitCollection> hfrh;
-  iEvent.getByLabel("hfreco", hfrh);
+  iEvent.getByToken(hfRecHitToken_, hfrh);
 
   edm::Handle<HORecHitCollection> horh;
-  iEvent.getByLabel("horeco", horh);
+  iEvent.getByToken(hoRecHitToken_, horh);
 
   if (hbhe.isValid()) {
     //    LogInfo("EcalCosmicHists") << "event " << ievt << " HBHE RecHits collection size " << hbhe->size();
```
Comment (tvami) [inline CaloOnlineTools/EcalTools/plugins/EcalCosmicsHists.cc:1269]: Please remove commented out code

File: CaloOnlineTools/EcalTools/plugins/EcalMipGraphs.cc [hunk 4]
```diff
@@ -114,21 +119,21 @@ void EcalMipGraphs::analyze(edm::Event const& iEvent, edm::EventSetup const& iSe
...
+  iEvent.getByToken(ebRecHitToken_, EBhits);
+  iEvent.getByToken(eeRecHitToken_, EEhits);
   // Now, retrieve the crystal digi from the event
-  iEvent.getByLabel(EBDigis_, EBdigisHandle);
-  iEvent.getByLabel(EEDigis_, EEdigisHandle);
+  iEvent.getByToken(ebDigiToken_, EBdigisHandle);
+  iEvent.getByToken(eeDigiToken_, EEdigisHandle);
   //debug
   //LogWarning("EcalMipGraphs") << "event " << ievt << " EBhits collection size " << EBhits->size();
   //LogWarning("EcalMipGraphs") << "event " << ievt << " EEhits collection size " << EEhits->size();
```
Comment (tvami) [inline CaloOnlineTools/EcalTools/plugins/EcalMipGraphs.cc:133]: Please remove commented out code

**Review by tvami — COMMENTED** [review]
Please remove commented out code

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca

**Comment by qliphy** [comment]
+1

**Pull request merged**